runtime/compiler/ilgen/Walker.cpp

```diff
@@ -2357,7 +2357,8 @@ TR_J9ByteCodeIlGenerator::genCheckCast()
  * Generate IL for a checkcast bytecode instruction.
  *
  * If the class specified in the bytecode is unresolved, this leaves out the
- * ResolveCHK since it has to be conditional on a non-null object.
+ * ResolveCHK since it has to be conditional on a non-null object when it is
+ * not a value type.
```

Comment:
I suggest you add explicitly that the class has to be resolved unconditionally if it's a value type, regardless of whether the value is null, just to make it clear.